src/wallet/rpcwallet.cpp

```diff
@@ -3770,13 +3781,14 @@ UniValue z_sendmany(const UniValue& params, bool fHelp)
         bool isZaddr = false;
         CTxDestination taddr = keyIO.DecodeDestination(address);
         if (!IsValidDestination(taddr)) {
-            auto res = keyIO.DecodePaymentAddress(address);
-            if (IsValidPaymentAddress(res)) {
+            auto decoded = keyIO.DecodePaymentAddress(address);
+            if (decoded.has_value()) {
+                libzcash::PaymentAddress addr(decoded.value());
                 isZaddr = true;
 
-                bool toSapling = std::get_if<libzcash::SaplingPaymentAddress>(&res) != nullptr;
-                bool toSprout = !toSapling;
-                noSproutAddrs = noSproutAddrs && toSapling;
+                bool toSapling = std::holds_alternative<libzcash::SaplingPaymentAddress>(addr);
+                bool toSprout = std::holds_alternative<libzcash::SproutPaymentAddress>(addr);
+                noSproutAddrs = !toSprout && noSproutAddrs && toSapling;
```

Comment:
This diff does not alter the semantics, but only because the current semantics are not correct. The addition of UAs to the `PaymentAddress` type meant that if a UA was specified as a recipient, `noSproutAddrs` would end up `true`. This change preserves that effect, but we will need to return to this logic to fix it up later (outside this PR).